src/ngraph/op/broadcast.cpp

```diff
@@ -166,10 +191,10 @@ void op::v1::Broadcast::validate_and_infer_types()
                 input_value(2).get_node_shared_ptr()->is_constant())
             {
                 auto target_shape =
-                    static_pointer_cast<op::Constant>(input_value(1).get_node_shared_ptr())
+                    static_pointer_cast<op::v0::Constant>(input_value(1).get_node_shared_ptr())
```

Comment:
`static_pointer_cast` means you are absolutely sure this is a Constant; if it isn't, it will give you garbage.
It looks like there are a lot of tests for the same value being a constant. Why not do one `as_type_ptr` at the beginning and just use that everywhere if it's not null?

Reply:
Done